### PR TITLE
[RW-7777][risk=no] Add missing access test for CohortReviewController

### PR DESCRIPTION
**Pull request by chenchals**
Description:
Add missing access test for CohortReviewController in the following test class CohortReviewControllerTest:
- CohortReviewController#createCohortReview :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#updateCohortReview :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#deleteCohortReview :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#createParticipantCohortAnnotation :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#updateParticipantCohortAnnotation :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#deleteParticipantCohortAnnotation :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#updateParticipantCohortStatus :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantCohortAnnotations :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantCohortStatus :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantCohortStatuses :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantCount :check_mark: +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantData :check_mark: +tests for WorkspaceAccessLevel
- CohortReviewController#getCohortReviewsInWorkspace :check_mark: refactored, +tests for WorkspaceAccessLevel
- CohortReviewController#getCohortChartData :check_mark:  +tests for WorkspaceAccessLevel
- CohortReviewController#getParticipantChartData :check_mark: +tests for WorkspaceAccessLevel
- CohortReviewController#getVocabularies :check_mark: +tests for WorkspaceAccessLevel

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 1]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
-        .thenReturn(workspace);
+  public void updateCohortReviewNoEtag() {
+    stubWorkspaceAccessLevel(workspace2, WorkspaceAccessLevel.WRITER);
 
-    stubBigQueryCohortCalls();
+    CohortReview requestCohortReview =
+        new CohortReview()
+            .cohortReviewId(cohortReview.getCohortReviewId())
+            .cohortId(cohortReview.getCohortId())
+            .etag(null);
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:691]: No need to set etag to null, It should be null on creation
```suggestion
    CohortReview requestCohortReview =
        new CohortReview()
            .cohortReviewId(cohortReview.getCohortReviewId())
            .cohortId(cohortReview.getCohortId());
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 2]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    participantId,
+                    request));
+    assertForbiddenException(exception);
+  }
+
+  ////////// updateParticipantCohortAnnotation  //////////
+  @Test
+  public void updateParticipantCohortAnnotationNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.WRITER);
+    Long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1044]: What is this sorcery? 
```suggestion
    Long wrongReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 3]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    participantAnnotation.getAnnotationId(),
+                    new ModifyParticipantCohortAnnotationRequest().annotationValueString("test1")));
+
+    assertNotFoundExceptionCohortReview(cohortReviewId, exception);
+  }
+
+  @Test
+  public void updateParticipantCohortAnnotationNoParticipantCohortStatus() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.WRITER);
+    Long participantId = participantCohortStatus1.getParticipantKey().getParticipantId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1063]:
```suggestion
    Long wrongParticipantId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 4]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    participantId,
+                    participantAnnotation.getAnnotationId(),
+                    new ModifyParticipantCohortAnnotationRequest().annotationValueString("test1")));
+
+    assertNotFoundExceptionParticipantCohortStatus(cohortReviewId, participantId, exception);
+  }
+
+  @Test
+  public void updateParticipantCohortAnnotationNoAnnotation() {
+    long badAnnotationId = participantAnnotation.getAnnotationId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1082]:
```suggestion
    Long badAnnotationId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 5]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
-          .isEqualTo(
-              "Not Found: No participant cohort annotation found for annotationId: "
-                  + annotationId
-                  + ", cohortReviewId: "
-                  + cohortReview.getCohortReviewId()
-                  + ", participantId: "
-                  + participantId);
-    }
+    DbParticipantCohortAnnotation annotation = saveTestParticipantCohortAnnotation();
+    Long annotationId = annotation.getAnnotationId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1261]:
```suggestion
    Long wrongAnnotationId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 6]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    participantCohortStatus1.getParticipantKey().getParticipantId(),
+                    new ModifyCohortStatusRequest().status(CohortStatus.INCLUDED)));
+
+    assertNotFoundExceptionNoCohortReviewAndCohort(cohortReview.getCohortReviewId(), exception);
+  }
+
+  @Test
+  public void updateParticipantCohortStatusNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.WRITER);
+    Long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1356]:
```suggestion
    Long wrongCohortReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 7]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    new ModifyCohortStatusRequest().status(CohortStatus.INCLUDED)));
+
+    assertNotFoundExceptionCohortReview(cohortReviewId, exception);
+  }
+
+  @Test
+  public void updateParticipantCohortStatusNoParticipantId() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.WRITER);
+    Long cohortReviewId = cohortReview.getCohortReviewId();
+    Long participantId = participantCohortStatus1.getParticipantKey().getParticipantId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1375]:
```suggestion
    Long wrongParticipantId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 8]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
   }
 
   @Test
-  public void getParticipantCohortAnnotations() {
-    when(workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
-            WORKSPACE_NAMESPACE, WORKSPACE_NAME, WorkspaceAccessLevel.READER))
-        .thenReturn(workspace);
+  public void getParticipantCohortAnnotationsNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.READER);
+    long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1457]:
```suggestion
    Long wrongCohortReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 9]
```diff
@@ -440,464 +539,1036 @@ public void setUp() {
...
+                    cohortReview.getCohortReviewId(),
+                    participantCohortStatus1.getParticipantKey().getParticipantId()));
+
+    assertNotFoundExceptionNoCohortReviewAndCohort(cohortReview.getCohortReviewId(), exception);
+  }
+
+  @Test
+  public void getParticipantCohortStatusNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.READER);
+    long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1544]:
```suggestion
    Long wrongCohortReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 10]
```diff
@@ -915,201 +1586,745 @@ public void getParticipantCohortStatus() {
...
+    assertNotFoundExceptionNoCohortReviewAndCohort(cohortReview.getCohortReviewId(), exception);
+  }
 
-    when(workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
-            WORKSPACE_NAMESPACE, WORKSPACE_NAME, WorkspaceAccessLevel.READER))
-        .thenReturn(workspace);
+  @Test
+  public void getParticipantCountNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.READER);
+    long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1632]:
```suggestion
    Long wrongCohortReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 11]
```diff
@@ -915,201 +1586,745 @@ public void getParticipantCohortStatus() {
...
+                    new PageFilterRequest().domain(Domain.CONDITION)));
+
+    assertNotFoundExceptionNoCohortReviewAndCohort(cohortReview.getCohortReviewId(), exception);
+  }
 
-    assertParticipantCohortStatuses(expectedReview4, null, null, null, null);
+  @Test
+  public void getParticipantDataNoCohortReview() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.READER);
+    long cohortReviewId = cohortReview.getCohortReviewId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1765]:
```suggestion
    Long wrongCohortReviewId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 12]
```diff
@@ -915,201 +1586,745 @@ public void getParticipantCohortStatus() {
...
-                cohortReview.getCohortReviewId(),
-                participantCohortStatus1.getParticipantKey().getParticipantId(),
-                new ModifyCohortStatusRequest().status(CohortStatus.INCLUDED))
-            .getBody();
-
-    assertThat(participantCohortStatus.getStatus()).isEqualTo(CohortStatus.INCLUDED);
+  public void getParticipantCohortStatusesNoCohort() {
+    stubWorkspaceAccessLevel(workspace, WorkspaceAccessLevel.READER);
+
+    Long cohortId = cohort.getCohortId() + 99L;
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1900]:
```suggestion
    Long cohortId = -1L;
```

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java [hunk 13]
```diff
@@ -1144,6 +2376,23 @@ private void stubBigQueryCohortCalls() {
...
+    // chart data - 0-string, 1-long, 2-long
+    when(bigQueryService.getString(null, 0)).thenReturn("1");
+    when(bigQueryService.getLong(null, 1)).thenReturn(1L);
+    // participant chart data - 0-string, 1-string, 2-date, 3-long, 4-long
+    when(bigQueryService.getDate(null, 2)).thenReturn("2000-01-01");
+    // vocabularies 0-string, 1-string, 2-string
+    when(bigQueryService.getString(null, 2)).thenReturn("1");
+    //         .domain(bigQueryService.getString(row, rm.get("domain")))
+    //        .type(bigQueryService.getString(row, rm.get("type")))
+    //        .vocabulary(bigQueryService.getString(row, rm.get("vocabulary"))));
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:2394]: Should these line be deleted?

**Review by freemabd — COMMENTED** [review]
Looks good. Just some nit comments.

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 13 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:2394]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 12 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1900]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 11 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1765]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 10 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1632]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 9 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1544]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 2 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1044]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 7 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1375]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 8 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1457]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java [hunk 14]
```diff
@@ -284,6 +284,7 @@ public void setUp() {
     TestMockFactory.stubCreateBillingProject(fireCloudService);
     TestMockFactory.stubCreateFcWorkspace(fireCloudService);
     TestMockFactory.stubPollCloudBillingLinked(cloudBillingClient, "billing-account");
+
```
[the remark this answers is not in the document]
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java:287]: removed empty line

File: api/src/test/resources/application.properties [hunk 15]
```diff
@@ -26,3 +26,13 @@ spring.jpa.properties.hibernate.show_sql=false
...
 #logging.level.org.hibernate.type.descriptor.sql.BasicBinder=TRACE
+
+# Run junit jupiter tests in parallel
+# see https://www.baeldung.com/junit-5-parallel-tests
+#junit.jupiter.execution.parallel.enabled = true
+#junit.jupiter.execution.parallel.mode.default = concurrent
+#junit.jupiter.execution.parallel.mode.classes.default = concurrent
+#junit.jupiter.execution.parallel.config.strategy=fixed
+#junit.jupiter.execution.parallel.config.fixed.parallelism=8
+#
```
[the remark this answers is not in the document]
Comment (chenchals) [reply api/src/test/resources/application.properties:37]: No not fully tested. removing these lines

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 1 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:691]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 3 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1063]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 5 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1261]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 4 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1082]: done

File: api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java (hunk 6 above)
Comment (chenchals) [reply api/src/test/java/org/pmiops/workbench/api/CohortReviewControllerTest.java:1356]: done

**Review by freemabd — APPROVED** [review]
Thanks.

**Pull request merged**